### PR TITLE
add discount amount attribute to bundle and configurable products

### PR DESCRIPTION
**Pull request by placiu**

### CODE REVIEW

File: Plugin/Catalog/Model/Indexer/DataProvider/Product/BundleDataExtender.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace CodingMice\VsBridgeIndexerExtension\Plugin\Catalog\Model\Indexer\DataProvider\Product;
+
+use Divante\VsbridgeIndexerCatalog\Model\Indexer\DataProvider\Product\BundleOptionsData;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\App\ObjectManager;
+
+class BundleDataExtender
+{
+    public function afterAddData(BundleOptionsData $subject, $indexData, $data, $storeId)
```
Comment (janmyszkier) [inline Plugin/Catalog/Model/Indexer/DataProvider/Product/BundleDataExtender.php]: \Divante\VsbridgeIndexerCatalog\Model\Indexer\DataProvider\Product\BundleOptionsData::addData
does not define that many arguments, see
```
    public function addData(array $indexData, $storeId)
```
meaning you have one too many

File: Plugin/Catalog/Model/Indexer/DataProvider/Product/ConfigurableDataExtender.php [hunk 2]
```diff
@@ -402,6 +404,26 @@ private function addHreflangUrls($indexData)
         return $indexData;
     }
 
+    private function addDiscountAmount($indexData, $storeId)
+    {
+        $objectManager = ObjectManager::getInstance();
```
Comment (janmyszkier) [inline Plugin/Catalog/Model/Indexer/DataProvider/Product/ConfigurableDataExtender.php]: please move this to class property, no need to instantiate object manager multiple times

File: Plugin/Catalog/Model/Indexer/DataProvider/Product/ConfigurableDataExtender.php [hunk 3]
```diff
@@ -402,6 +404,26 @@ private function addHreflangUrls($indexData)
...
+        $productRepository = $objectManager->create(ProductRepositoryInterface::class);
+
+        foreach ($indexData as $product_id => $indexDataItem) {
+            $productTypeID = $indexData[$product_id]['type_id'];
+            if ($productTypeID == 'simple' || $productTypeID == 'bundle') {
```
Comment (janmyszkier) [inline Plugin/Catalog/Model/Indexer/DataProvider/Product/ConfigurableDataExtender.php]: is this right? the configurable children are `simple` type. either this is wrong or you didn't add the loop for configurable children.

File: Plugin/Catalog/Model/Indexer/DataProvider/Product/BundleDataExtender.php (hunk 1 above)
Comment (placiu) [reply Plugin/Catalog/Model/Indexer/DataProvider/Product/BundleDataExtender.php]: @janmyszkier Well according to the docs after methods get subject, result and have access to all the arguments of their observed methods. I've struggle with this when I was var_dumping $storeId and receiving $indexData. So: $indexData is a result, $data is original $indexData.
I will rename those variables.

**Pull request merged**